Use CheckedPtr / CheckedRef less for non-stack objects

cdumez opened the pull request:
#### 98667dc58f6ddf80965e72e370be7a878e89122d
<pre>
Use CheckedPtr / CheckedRef less for non-stack objects
<a href="https://bugs.webkit.org/show_bug.cgi?id=266503">https://bugs.webkit.org/show_bug.cgi?id=266503</a>

Reviewed by Darin Adler and Ryosuke Niwa.

* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityScrollView.h:
* Source/WebCore/loader/ApplicationManifestLoader.h:
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::ApplicationCacheHost):
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/WebCore/page/DOMWindowExtension.h:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::widgetForElement):
(WebCore::EventHandler::completeWidgetWheelEvent):
(WebCore::EventHandler::handleWheelEventInternal):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::didAddWidgetToRenderTree):
(WebCore::LocalFrameView::willRemoveWidgetFromRenderTree):
(WebCore::collectAndProtectWidgets):
(WebCore::LocalFrameView::updateWidgetPositions):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::absoluteEventTrackingRegionsForFrame const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::ScrollbarThemeMac::registerScrollbar):
(WebCore::ScrollbarThemeMac::unregisterScrollbar):
* Source/WebCore/platform/mac/ScrollbarsControllerMac.mm:
* Source/WebCore/rendering/CounterNode.cpp:
(WebCore::CounterNode::~CounterNode):
(WebCore::CounterNode::nextInPreOrderAfterChildren const):
(WebCore::CounterNode::lastDescendant const):
(WebCore::CounterNode::previousInPreOrder const):
(WebCore::CounterNode::resetThisAndDescendantsRenderers):
(WebCore::CounterNode::recount):
(WebCore::CounterNode::insertAfter):
(WebCore::CounterNode::removeChild):
* Source/WebCore/rendering/CounterNode.h:
* Source/WebCore/rendering/RenderCounter.cpp:
(WebCore::makeCounterNode):
(WebCore::RenderCounter::originalText const):
* Source/WebCore/rendering/RenderCounter.h:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleDOMWindowExtension.cpp:
(WebKit::InjectedBundleDOMWindowExtension::InjectedBundleDOMWindowExtension):
(WebKit::InjectedBundleDOMWindowExtension::~InjectedBundleDOMWindowExtension):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleDOMWindowExtension.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.cpp:
(WebKit::overlayMap):
(WebKit::WebPageOverlay::WebPageOverlay):
(WebKit::WebPageOverlay::~WebPageOverlay):
(WebKit::WebPageOverlay::fromCoreOverlay):
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.h:

Canonical link: <a href="https://commits.webkit.org/272165@main">https://commits.webkit.org/272165@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3685f97aed70b1c7c08f43bfc74812a0dcb1dcd9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30791 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9470 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32459 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33287 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27822 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31536 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11778 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6713 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27693 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31105 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7944 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27544 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6816 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6964 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27420 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34624 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28017 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27901 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33120 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7009 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5088 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30953 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8726 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7726 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3991 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7565 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -3160,7 +3160,7 @@ HandleUserInputEventResult EventHandler::handleWheelEventInternal(const Platform
                     return *wheelEventDataForRemoteFrame;
-            } else if (WeakPtr<Widget> widget = widgetForElement(*element)) {
+            } else if (RefPtr widget = widgetForElement(*element)) {
                 if (passWheelEventToWidget(event, *widget.get(), processingSteps))
```
No need for the `get()` now that it’s a `RefPtr`.

darinadler reviewed (approved):
Is it safe to use weak pointers as hash table keys?

cdumez committed:
Use CheckedPtr / CheckedRef less for non-stack objects
https://bugs.webkit.org/show_bug.cgi?id=266503

Reviewed by Darin Adler and Ryosuke Niwa.

* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityScrollView.h:
* Source/WebCore/loader/ApplicationManifestLoader.h:
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::ApplicationCacheHost):
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/WebCore/page/DOMWindowExtension.h:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::widgetForElement):
(WebCore::EventHandler::completeWidgetWheelEvent):
(WebCore::EventHandler::handleWheelEventInternal):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::didAddWidgetToRenderTree):
(WebCore::LocalFrameView::willRemoveWidgetFromRenderTree):
(WebCore::collectAndProtectWidgets):
(WebCore::LocalFrameView::updateWidgetPositions):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::absoluteEventTrackingRegionsForFrame const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::ScrollbarThemeMac::registerScrollbar):
(WebCore::ScrollbarThemeMac::unregisterScrollbar):
* Source/WebCore/platform/mac/ScrollbarsControllerMac.mm:
* Source/WebCore/rendering/CounterNode.cpp:
(WebCore::CounterNode::~CounterNode):
(WebCore::CounterNode::nextInPreOrderAfterChildren const):
(WebCore::CounterNode::lastDescendant const):
(WebCore::CounterNode::previousInPreOrder const):
(WebCore::CounterNode::resetThisAndDescendantsRenderers):
(WebCore::CounterNode::recount):
(WebCore::CounterNode::insertAfter):
(WebCore::CounterNode::removeChild):
* Source/WebCore/rendering/CounterNode.h:
* Source/WebCore/rendering/RenderCounter.cpp:
(WebCore::makeCounterNode):
(WebCore::RenderCounter::originalText const):
* Source/WebCore/rendering/RenderCounter.h:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleDOMWindowExtension.cpp:
(WebKit::InjectedBundleDOMWindowExtension::InjectedBundleDOMWindowExtension):
(WebKit::InjectedBundleDOMWindowExtension::~InjectedBundleDOMWindowExtension):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleDOMWindowExtension.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.cpp:
(WebKit::overlayMap):
(WebKit::WebPageOverlay::WebPageOverlay):
(WebKit::WebPageOverlay::~WebPageOverlay):
(WebKit::WebPageOverlay::fromCoreOverlay):
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.h:

Canonical link: https://commits.webkit.org/272165@main
